### PR TITLE
Add option to display raw API definitions

### PR DESCRIPTION
**Pull request by Fox32**
While we have the technical definition of the APIs, there is no way to access it when a viewer is available. This adds a way to the API card to access the raw definition.

This also adds an option to add a copy button to code snippets.

It is also now possible to configure the definition types with own components to support custom viewers. This conflicts a bit with #2088, but should be easy to merge.

image

image

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: packages/core/src/components/CodeSnippet/CodeSnippet.test.tsx [hunk 1]
```diff
@@ -55,4 +55,14 @@ describe('<CodeSnippet />', () => {
...
     expect(queryByText(/3/)).toBeInTheDocument();
   });
+
+  it('copy code using button', async () => {
+    document.execCommand = jest.fn();
+    const rendered = render(
+      wrapInTestApp(<CodeSnippet {...minProps} showCopyCodeButton />),
+    );
+    const button = rendered.getByTitle('Text copied to clipboard');
+    button.click();
```
Comment (freben) [inline packages/core/src/components/CodeSnippet/CodeSnippet.test.tsx:65]: we're supposed to use `fireEvent.click(button)`, or even the newfangled `userEvent.click(button)`, aren't we?

**Review by freben — APPROVED** [review]
Looks good! Just one open question

File: packages/core/src/components/CodeSnippet/CodeSnippet.test.tsx (hunk 1 above)
Comment (Fox32) [reply packages/core/src/components/CodeSnippet/CodeSnippet.test.tsx:65]: Copied that test from somewhere else, but I will take a look and may fix it at both places

Comment (Fox32) [reply packages/core/src/components/CodeSnippet/CodeSnippet.test.tsx]: Switched to `fireEvent` as I couldn't find any uses of `userEvent`

Comment (freben) [reply packages/core/src/components/CodeSnippet/CodeSnippet.test.tsx]: Makes sense! The userEvent library is pretty new and we haven't adopted it yet.

**Comment by freben** [comment]
Go ahead and merge, you should be able to :)

**Pull request merged**